pkg/proxy/util/nodeport_addresses.go

```diff
@@ -82,20 +97,20 @@ func GetNodeAddresses(cidrs []string, nw NetworkInterfacer) (sets.String, error)
 	}
 
 	if uniqueAddressList.Len() == 0 {
-		return nil, fmt.Errorf("no addresses found for cidrs %v", cidrs)
+		return nil, fmt.Errorf("no addresses found for cidrs %v", npa.cidrStrings)
 	}
 
 	return uniqueAddressList, nil
 }
 
-// ContainsIPv4Loopback returns true if the input is empty or one of the CIDR contains an IPv4 loopback address.
-func ContainsIPv4Loopback(cidrStrings []string) bool {
-	if len(cidrStrings) == 0 {
+// ContainsIPv4Loopback returns true if npa's CIDRs contain an IPv4 loopback address.
+func (npa *NodePortAddresses) ContainsIPv4Loopback() bool {
```

Comment:
nit, since `cidrStrings` is readonly you can optimize this by caching the result in a bool, probably initialize it  in `NewNodePortAddresses`

Reply:
nevermind, happens in the next commit